### ISSUE TITLE
Need a textured 3D .obj model for the DualShock4 controller

### ISSUE THREAD

**Issue by HipsterSloth** [cboulay/PSMoveService#150]
This was spawned off of issue #41. 

I need a 3d model (.obj format) with single texture for the dual shock 4 controller to display in SteamVR and in the config tool. I'm currently using the generic steamvr controller model as a stand in.

In SteamVR the model units are in meters. Once I have a. OBJ model for SteamVR I can add it into the config tool since there is a special process involved in doing that (need to run a perl script to turn the .obj into a c++ header file).

**Comment by andregm3** [comment]
https://sketchfab.com/models/61080a101b0b4471b4e10d5991525067

**Comment by HipsterSloth** [comment]
@andregm3 Unless I'm mistaken (and I may be, I've never used sketchfab before) this model doesn't appear to be available for download. It appears to be a display only model. I created a SketchFab account and I still didn't see a download option appear.

**Comment by andregm3** [comment]
yeah i will try to find more tonight

**Comment by andregm3** [comment]
http://tf3dm.com/download-page.php?url=gamepad-keyboard-hybrid-59481

https://grabcad.com/library/dual-shok-4-play-station-1

**Comment by andregm3** [comment]
https://grabcad.com/library/dualshock-4-for-playstation-1

**Comment by HipsterSloth** [comment]
> https://grabcad.com/library/dualshock-4-for-playstation-1

Generally this looks good, but I have some thoughts:
* This model looks like it should work as far as the geometry and licensing is concerned (I gave a quick look over the terms of usage). 
* Unfortunately it doesn't have an textures included with it. I can only open the STL version of the model since I only have Blender, so it's possible that one of the other formats (Catia, creo3, Parasolid, solidworks, Step) might have the textures baked in, but looking at the folders it doesn't look to be the case. I'm terrible at texturing so I could use some help texturing this model. 
* This model is a bit on the high-poly side. If someone who knew how to reasonably simply the mesh do sub 30k verts that would be awesome. This is an issue because I bake the model into the config tool executable and large models will slow down the compile time.

@andregm3 Thanks for researching these!

**Comment by andregm3** [comment]
https://github.com/cboulay/PSMoveService/files/412021/dualshock_4.zip

**Comment by InfinateXtremer** [comment]
Hey i'm back, sorry for going offline, had too much stuff happen on my personal life, also my main hdd died but i'm back :D. From looks of it i can optimize model from grabcat and do basic uv map.

**Comment by HipsterSloth** [comment]
@InfinateXtremer Oh no worries! I know people have a lot going on in the summer, especially in August. Sorry to hear about the HDD. Thanks for taking a look at this.

**Comment by InfinateXtremer** [comment]
I optimized and recreated some parts of the controller, i made 2 models one is 16,041 verts and the other is 18,914 verts, both of them have edge split applied. I can go lower polycount if you want but it's probably going to look like blocky. Original model's vert count is 40,515.

**Comment by InfinateXtremer** [comment]
Okay i reduced more verts it's 12k now. sadly i still haven't done uv mapping but actually assigned every texture to vertex and now all i need to do this unwrap uv map, assign right textures and bake them all.

Here is current version if you wanna test compile time.
https://dl.dropboxusercontent.com/u/70822812/ds4.blend

**Comment by HipsterSloth** [comment]
@InfinateXtremer Thank you for all your work on that! Much appreciated. I'll try and do a compile test by no later than this weekend. I'm currently heads down trying to get some DS4 tracking work done. As soon as I have a moment to get this integrated into the build I'll let you know. That said I suspect 12k verts should be fine.

**Comment by InfinateXtremer** [comment]
So i was able to map it and here it is https://dl.dropboxusercontent.com/u/70822812/DS4MAP.zip it's not done 100% but it should work without major texture or mesh bugs.

**Comment by HipsterSloth** [comment]
@InfinateXtremer Thanks again for your work on this! I did a bit of work today getting the model imported into the config tool and wanted to let you know about some issues I came across. I rotated the model a bit to align the center of the light bar (for tracking reasons the lightbar defines the origin). I couldn't get the texture to show up in blender when I first opened the .blend file. I was able to work around this by modifying the mtl file to point at the texture image and re-import the obj. I was able to get the model showing up in the config tool (build time was fine with 12k verts):

![image](https://cloud.githubusercontent.com/assets/16793853/18228312/007fce86-71fe-11e6-97a5-a7f314da0cea.png)

I think some parts aren't rendering because there are polygons on the model that aren't triangles. The light bar was one such piece. I need all polygons to be triangles so that I can render the controller as one triangle list. All the source geometry I tweaked has been checked in here:

https://github.com/cboulay/PSMoveService/tree/master/src/psmoveconfigtool/source_assets/DS4

**Comment by InfinateXtremer** [comment]
Yeah i definately forgot to triangulate mesh,. also looking at your picture i probably need to tweak the colors and add ambient occlusion to texture.

**Comment by InfinateXtremer** [comment]
I triangulated mesh, tweaked colors little bit and added ambient occlusion, i also included unused textures just in case https://dl.dropboxusercontent.com/u/70822812/ds4new.rar
